File: CLLIX.TAAuditTracker.Domain/BookingReservation.cs
namespace CLLIX.TAAuditTracker.Domain
{
    public class BookingReservation : BaseEntity
    {
        public string ApartmentPropertyName { get; set; } = string.Empty; // FK        

        public string InvoiceNumber { get; set; } = string.Empty;
        public string StatementNumber { get; set; } = string.Empty;
        public string TravelAgentBilling { get; set; } = string.Empty;

        public string ReservationNumber { get; set; } = string.Empty;
        public string ConfirmationNumber { get; set; } = string.Empty;

        public DateTime CheckInDate { get; set; }
        public DateTime CheckOutDate { get; set; }
        public int Nights { get; set; }

        public string GuestName { get; set; } = string.Empty;
        public string TravelAgentName { get; set; } = string.Empty;
        public string BookingSource { get; set; } = string.Empty;

        public string CommissionRate { get; set; } = string.Empty; // e.g. "10%" or "12%"
        public decimal DailyTariff { get; set; }
        public decimal TotalTariff { get; set; }
        public decimal TotalCommission { get; set; }
        public decimal AmountInTAInvoice { get; set; }
        public bool IsInvoiceMatched { get; set; }
        public string InvoiceRemarks { get; set; } = string.Empty;
        public int? WeekNumber { get; set; }

        public DateTime? InvoiceReceivedDate { get; set; }
        public DateTime? InvoiceProcessDate { get; set; }
        public DateTime? DueDate { get; set; }
        public DateTime? RemittanceDate { get; set; }

        public bool IsRemitted { get; set; }
        public string Status { get; set; } = string.Empty;
        public string Remarks { get; set; } = string.Empty;

        // Optional: Navigation property to TravelAgency
        public int? TravelAgencyId { get; set; }
        public TravelAgency? TravelAgency { get; set; }
    }
}

File: CLLIX.TAAuditTracker.ClientUi/src/app/_interfaces/booking-reservation.ts
import { BaseEntity } from './base-entity';
import { TravelAgency } from './travel-agency';

export interface BookingReservation extends BaseEntity {
  apartmentPropertyName: string;

  invoiceNumber: string;
  statementNumber: string;
  travelAgentBilling: string;

  reservationNumber: string;
  confirmationNumber: string;

  checkInDate: string;
  checkOutDate: string;
  nights: number;

  guestName: string;
  travelAgentName: string;
  bookingSource: string;

  commissionRate: string;
  dailyTariff: number;
  totalTariff: number;
  totalCommission: number;
  amountInTAInvoice: number;
  isInvoiceMatched: boolean;
  invoiceRemarks: string;
  weekNumber?: number;

  invoiceReceivedDate?: string;
  invoiceProcessDate?: string;
  dueDate?: string;
  remittanceDate?: string;

  isRemitted: boolean;
  status: string;
  remarks: string;

  travelAgencyId?: number;
  travelAgency?: TravelAgency;
}
